File: Problems & Solutions/45. Jump Game II/08 February 2023/Approach 1 - Greedy/Jump Game II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int answer = 0, n = int(nums.size());

        int curEnd = 0, curFar = 0;
        
        for (int i = 0; i < n - 1; ++i) {
            curFar = max(curFar, i + nums[i]);

            if (i == curEnd) {
                answer++;
                curEnd = curFar;
            }
        }
        
        return answer;
    }
};

File: Problems & Solutions/45. Jump Game II/08 February 2023/Approach 1 - Greedy/Jump Game II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        answer, n = 0, len(nums)
        cur_end, cur_far = 0, 0
        
        for i in range(n - 1):
            cur_far = max(cur_far, i + nums[i])

            if i == cur_end:
                answer += 1
                cur_end = cur_far
                
        return answer